### PR TITLE
FI-2481 Add PractitionerRole

### PR DESCRIPTION
**Pull request by yunwwang**
# Summary
This fixed JIRA ticket FI-2481

# Change Log
* Add PracitionerRole test to US Core v5.0.1 and above
* Add Location test to US Core v7.0.0-ballot
* Remove Vital Sign test from US Core v7.0.0-ballot
* Remove duplicated SMART granular test from generated folder
* 

# Testing Guidance
Start US Core Test Kit
Verify US Core v5.0.1 test suite has optional PracitionerRole test group
Verify US Core v6.1.0 test suite has optional PracitionerRole test group
Verify US Core v7.0.0 ballot test suite has optional PracitionerRole test group
Verify US Core v7.0.0 ballot test suite has Location test group
Verify US Core v7.0.0 ballot test suite does NOT have VitalSign test group

### CODE REVIEW

File: lib/us_core_test_kit/generator/ig_metadata_extractor.rb [hunk 1]
```diff
@@ -52,10 +53,17 @@ def add_missing_supported_profiles
...
+        resources_in_capability_statement.each do |resource|
+          resource.supportedProfile.map! { |profile_url| profile_url.split('|').first }
+        end
+      end
+
       def remove_extra_supported_profiles
         ig_resources.capability_statement.rest.first.resource
             .find { |resource| resource.type == 'Observation' }
             .supportedProfile.delete_if do |profile_url|
+              binding.pry if ig_resources.ig.version == '7.0.0-ballot' && profile_url.include?('vital-signs')
```
Comment (Jammjammjamm) [inline lib/us_core_test_kit/generator/ig_metadata_extractor.rb:66]: Do we still need that pry? There's one in the `GroupMetadataExtractor` too.

**Review by Jammjammjamm — COMMENTED** [review]
This might require some major changes, and I don't know that we'd want to undertake this now, but I feel like the PractitionerRole tests could have `practitioner` as a first search, rather than being a read first group.

File: lib/us_core_test_kit/generator/special_cases.rb [hunk 2]
```diff
@@ -1,20 +1,29 @@
...
     module SpecialCases
-      RESOURCES_TO_EXCLUDE = [
-        'Location',
-        'Medication',
-        'PractitionerRole'
-      ].freeze
+      RESOURCES_TO_EXCLUDE = {
+        'Location' => ['v311', 'v400', 'v501', 'v610'],
+        'Medication' => ['v311', 'v400', 'v501', 'v610', 'v700_ballot'],
+        'PractitionerRole' => ['v311', 'v400']
```
[the remark this answers is not in the document]
Comment (yunwwang) [reply lib/us_core_test_kit/generator/special_cases.rb:7]: Because Reference to PractitionerRole is not MustSupport reference in v3 & v4

**Comment by yunwwang** [comment]
> This might require some major changes, and I don't know that we'd want to undertake this now, but I feel like the PractitionerRole tests could have `practitioner` as a first search, rather than being a read first group.

This is an interesting question. PractitionerRole.practitioner is a MustSupport and optional element. And PractitionerRole.practitioner SHALL support chained prameters. Hmm.

**Comment by yunwwang** [comment]
Also how we incoroprate this group to g10 is another big issue.

**Comment by Jammjammjamm** [comment]
> Also how we incoroprate this group to g10 is another big issue.

Is this currently set up to save PractitionerRole references like other delayed resources?

**Comment by yunwwang** [comment]
I have another ticket to update reference server data. PractitionerRole is treaty the same as other delayed reference, and the test group is currently marked as an optional group in US Core. If we want to include this test group in g10, then we have to remove the optional property. We need to figure out how to say that If there are delayed PractitionerRole reference, then run the PractitionerRole test group. If there is not delayed reference, then omit this test group. We need another property, other than optional. Maybe "semi-optional"?

**Review by Jammjammjamm — APPROVED** [review]

**Comment by Jammjammjamm** [comment]
> We need to figure out how to say that If there are delayed PractitionerRole reference, then run the PractitionerRole test group. If there is not delayed reference, then omit this test group. We need another property, other than optional. Maybe "semi-optional"?

I'll give that some thought. It's pretty annoying that this one needs to be treated slightly different than all the other groups.

**Pull request merged**